libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParser.java

```diff
@@ -97,6 +99,58 @@ public PolicyParser(InputStream inputStream, String policyName, boolean isExtern
         this.externalEntitlements = externalEntitlements;
     }
 
+    public VersionedPolicy parseVersionedPolicy() {
```

Comment:
My suggestion would be to not have the version parsing as part of the `PolicyParser` at all since it's only used to determine if a policy override is valid. 

I see two different possibilities to accomplish this:
1. Change `PolicyParser` to take in an XContentBuilder instead of an InputStream. Pull out the xcontent necessary for version checking in elasticsearch bootstrapping code.
2. Change the override to only be JSON for the policy and some stream of bytes for the versions. You can then read that stream of bytes up to where the policy begins and then pass the remainder of the `InputStream` into the `PolicyParser`.

I think this may also remove the need for a VersionedPolicy record as well.